File: java/src/test/java/io/binac/leetcode/LargestRectangleInHistogramTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LargestRectangleInHistogramTest {
    private static final LargestRectangleInHistogram.Solution1 SOLUTION1 = new LargestRectangleInHistogram.Solution1();

    private final int[] HEIGHTS1 = {2, 1, 5, 6, 2, 3};
    private final int EXPECTED1 = 10;

    private final int[] HEIGHTS2 = {2, 4};
    private final int EXPECTED2 = 4;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.largestRectangleArea(HEIGHTS1));
        assertEquals(EXPECTED2, SOLUTION1.largestRectangleArea(HEIGHTS2));
    }
}

File: cpp/tests/LargestRectangleInHistogramTest.cpp
#include "LargestRectangleInHistogram.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(LargestRectangleInHistogram, Solution84_1) {
  auto S84_1 = Solution84_1();

  std::vector<int> Heights1{2, 1, 5, 6, 2, 3};
  int const Expected1 = 10;
  EXPECT_EQ(Expected1, S84_1.largestRectangleArea(Heights1));

  std::vector<int> Heights2{2, 4};
  int const Expected2 = 4;
  EXPECT_EQ(Expected2, S84_1.largestRectangleArea(Heights2));
}